whisperv6/whisper.go

```diff
@@ -418,17 +420,17 @@ func (whisper *Whisper) SendHistoricMessageResponse(peer *Peer, payload []byte)
 }
 
 // SendP2PMessage sends a peer-to-peer message to a specific peer.
-func (whisper *Whisper) SendP2PMessage(peerID []byte, envelope *Envelope) error {
+func (whisper *Whisper) SendP2PMessage(peerID []byte, envelopes ...*Envelope) error {
 	p, err := whisper.getPeer(peerID)
 	if err != nil {
 		return err
 	}
-	return whisper.SendP2PDirect(p, envelope)
+	return whisper.SendP2PDirect(p, envelopes...)
 }
 
 // SendP2PDirect sends a peer-to-peer message to a specific peer.
-func (whisper *Whisper) SendP2PDirect(peer *Peer, envelope *Envelope) error {
-	return p2p.Send(peer.ws, p2pMessageCode, envelope)
+func (whisper *Whisper) SendP2PDirect(peer *Peer, envelopes ...*Envelope) error {
+	return p2p.Send(peer.ws, p2pMessageCode, envelopes)
```

Comment:
how do you now that client will be able to consume batch of envelopes? old clients will fail to decode this, right?

Reply:
it seems that if we want to make it backward compatible we will have to add new code, smth like requestHistoryBatch

Reply:
Oh yeah, you're right. I will change it to send it as a single envelope if number of items is 1.

But regarding the clients, I was thinking about adding an argument to the request indicating whether it supports batch processing or not.

Reply:
oh, i forgot that the other part of the change will be in status-go/mailserver module

Reply:
> I will change it to send it as a single envelope if number of items is 1.

good idea, that would work until we change the mail server part!